### DISCUSSION TITLE
Very vanilla script to import a JSON array into my database, getting a timeout error.

### DISCUSSION THREAD

**Discussion by sergiotapia** [prisma/prisma#16884] [Q&A]
I'm testing out Nextjs 13 and Prisma and this simple script just writes the Platforms into the Platform table.

```
// THIS IS IN: localhost:3000/api/igdb/refresh_platforms
import type { NextApiRequest, NextApiResponse } from "next";
import axios from "axios";
import prisma from "../../../lib/prisma";
interface Platform {
  id: number;
  abbreviation: string;
  alternative_name: string;
  category: number;
  created_at: number;
  generation: number;
  name: string;
  platform_logo: {
    id: number;
    alpha_channel: boolean;
    animated: boolean;
    height: number;
    image_id: string;
    url: string;
    width: number;
    checksum: string;
  };
  slug: string;
  summary: string;
  updated_at: number;
  url: string;
  versions: number[];
  checksum: string;
}

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  const { method } = req;

  if (method != "POST") {
    return res.status(404).json({ error: "Unknown route" });
  }

  const url = "https://api.igdb.com/v4/platforms";

  const headers = {
    "client-id": process.env.TWITCH_IGDB_CLIENT_ID,
    authorization: "Bearer my_token",
    "accept-encoding": "*",
  };

  const body = `fields *, platform_family.*, platform_logo.*;
  limit 500;`;

  const response = await axios.post(url, body, { headers });

  console.log("The response status is:", response.status);

  if (response.status == 200) {
    response.data.forEach(async (p: Platform) => {
      const platform = await prisma.platform
        .create({
          data: {
            category: p.category,
            name: p.name,
            slug: p.slug,
            platform_logo_url: p.platform_logo?.url, // Todo, sanitize this url.
            hashtag: p.name, // Todo, convert name to hashtag
          },
        });

      console.log("Created platform", platform);
    });
  } else {
    console.log(response.status);
  }

  res.status(200).json({ message: "Imported all platforms." });
}
```

The error:

> 2022-12-18T19:37:17.574Z	46bc23b6-c7b0-40d0-9d56-32fef1c700af	ERROR	Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: \nInvalid `prisma.platform.create()` invocation:\n\n\nTimed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 5)","reason":{"errorType":"Error","errorMessage":"\nInvalid `prisma.platform.create()` invocation:\n\n\nTimed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 5)","clientVersion":"4.7.1","stack":["Error: ","Invalid `prisma.platform.create()` invocation:","","","Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 5)","    at RequestHandler.handleRequestError (/var/task/node_modules/@prisma/client/runtime/index.js:34883:13)","    at RequestHandler.handleAndLogRequestError (/var/task/node_modules/@prisma/client/runtime/index.js:34841:12)","    at RequestHandler.request (/var/task/node_modules/@prisma/client/runtime/index.js:34836:12)","    at async PrismaClient._request (/var/task/node_modules/@prisma/client/runtime/index.js:35926:16)","    at async /var/task/.next/server/pages/api/igdb/refresh_platforms.js:77:30"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: Error: ","Invalid `prisma.platform.create()` invocation:","","","Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 5)","    at process.<anonymous> (file:///var/runtime/index.mjs:1194:17)","    at process.emit (node:events:525:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
> Unknown application error occurred
> Runtime.Unknown

Finally my `lib/prisma.ts` file:

```
import { PrismaClient } from "@prisma/client";
declare let global: { prisma: PrismaClient };

// PrismaClient is attached to the `global` object in development to prevent
// exhausting your database connection limit.
//
// Learn more:
// https://pris.ly/d/help/next-js-best-practices

let prisma: PrismaClient;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient({
      log: ["query"],
    });
  }
  prisma = global.prisma;
}
export default prisma;
```

This is as vanilla as can be from my outside eyes, so I don't know what I'm doing wrong. Appreciate any help!

**Comment by pantharshit00** [comment]
Hello @sergiotapia,

In a serverless environment if you application receives enough traffic, it can create multiple lambda functions with separate connections and that can exhaust the database connection limit. 

In order to resolve this, you can either set the connection_limit parameter in the connection string to `1` or you can use a connection pooler like [Prisma Data Proxy](https://www.prisma.io/data-platform/proxy) or [PgBouncer](https://www.pgbouncer.org/). 

You can read further about this in the documentation piece: https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#serverless-environments-faas

**Reply by sergiotapia** [reply]
I'm only hitting this URL once, and waiting for it to complete then refreshing my database viewer (tableplus). 

I'm not refreshing the url over and over.

**Reply by pantharshit00** [reply]
In that case, can you please share a minimal reproduction in a small git repository which I can try to deploy?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
